### DISCUSSION TITLE
Support more than 4 controllers

### DISCUSSION THREAD

**Discussion by Remixful** [godotengine/godot-proposals#5901] [General Discussions]
Hi, I'm working on a local multiplayer game for up to 8 players.
Currently, Godot will not detect more than 4 controllers (I'm assuming it's due to XInput)
I'm not too experienced on this subject or on how to fix it, but I've seen some games have the option to disable XInput to support up to 8 controllers, and I'd love to have a similar feature for Godot.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot/issues/19173.

XInput doesn't support more than 4 controllers, so the legacy DirectInput API would have to be used here. I'm afraid this would be a lot of work for little gain though.

**Comment by troybonneau** [comment]
I'm working on a local multiplayer game for up to 8 players in Godot, and I've run into the same issue. I disagree about the little gain in supporting this feature, as both Switch + XBOX natively support that amount and I have made games with Unity using Rewired and Gamemaker Studio that have no problem supporting more than 4 controllers. I also think the device menu having showing 8 devices will continue to bring people to this page because it's unclear that the engine doesn't support this.
 
 
![image](https://github.com/godotengine/godot-proposals/assets/53018809/6c196f7c-ff07-4897-b584-46dfec00e399)

 
 I think my workaround for the time being will be using some sort of remapping software such as joytokey.

**Comment by schemesmith** [comment]
Has anyone found a good workaround for at least testing a project with 8 controllers? Was joytokey the best solution? Thanks.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/8620.
